src/OctoPrint.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *  OpenSCAD (www.openscad.org)
+ *  Copyright (C) 2009-2019 Clifford Wolf <clifford@clifford.at> and
+ *                          Marius Kintel <marius@kintel.net>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#pragma once
+
+#include <QtGlobal>
+
+#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
+
+// Dummy class for compiling Preferences.cc for Qt4
+class OctoPrint
+{
+public:
+	OctoPrint() {}
+	~OctoPrint() {}
+
+	const std::tuple<std::string, std::string, std::string> get_version() const { return {}; }
+	const std::vector<std::pair<std::string, std::string>> get_slicers() const { return {}; }
+	const std::vector<std::pair<std::string, std::string>> get_profiles(const QString) const { return {}; }
+};
+
+#else
+
+#include <tuple>
+
+#include <QFile>
+#include <QString>
+#include <QJsonDocument>
+
+#include "Network.h"
+
+class OctoPrint
+{
+public:
+	OctoPrint();
+	virtual ~OctoPrint();
+
+	const QString url() const;
+	const std::string api_key() const;
```

Comment:
We should try to stick to camelCase. I know we're violating this in a bunch of places, so I'm happy to defer this to a later cleanup if it's annoying to deal with right now.

Reply:
done